### ISSUE TITLE
reaction_mongo-init-replica_1 fails to connect

### ISSUE THREAD

**Issue by janus-reith** [reactioncommerce/reaction-platform#16]
## Issue Description
After running make from the clean reaction-platform folder, the created containers `reaction_reaction_1`, `reaction_mongo-init-replica_1` and `reaction-hydra_hydra-migrate_1` show as exited.

The reaction container fails, as it can't connect to the replica set:
`docker logs -f reaction_reaction_1`

> npm WARN react-addons-test-utils@15.6.2 requires a peer of react-dom@^15.4.2 but none is installed. You must install peer dependencies yourself.
> npm WARN react-taco-table@0.5.1 requires a peer of react@^15.3 but none is installed. You must install peer dependencies yourself.
> npm WARN react-taco-table@0.5.1 requires a peer of react-dom@^15.3 but none is installed. You must install peer dependencies yourself.
> npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules/fsevents):
> npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
> 
> up to date in 18.481s
> ERROR: MongoDB replica set not ready in time.
> no replset config has been received..%  

The replica set fails to  connect to the main mongo container:
`docker logs -f reaction_mongo-init-replica_1`
```
MongoDB shell version v3.6.3
connecting to: mongodb://mongo:27017/reaction
2018-10-04T11:25:18.048+0000 W NETWORK  [thread1] Failed to connect to 172.30.0.2:27017, in(checking socket for error after poll), reason: Connection refused
2018-10-04T11:25:18.062+0000 E QUERY    [thread1] Error: couldn't connect to server mongo:27017, connection attempt failed :
connect@src/mongo/shell/mongo.js:251:13
@(connect):1:6
exception: connect failed
```

I checked the internal IP, it seems to be right
`docker inspect reaction_mongo_1 | grep IPAddress`

> "SecondaryIPAddresses": null,
>             "IPAddress": "",
>                     "IPAddress": "172.30.0.2",

## Steps to Reproduce
Please provide starting context, i.e. logged in as a user, configure a particular payment method.
1. `clone` the repo
2. run `make` and wait for it to finish
3. run `docker ps -a | grep reaction` and observe that both `reaction_reaction_1` and `reaction_mongo-init-replica_1` exited

## Possible Solution
Running `docker start reaction_mongo-init-replica_1`  afterwards seems to make it start - Maybe the build script needs to wait until the main mongodb container finished loading.

**Comment by spencern** [comment]
Thanks for the report @janus-reith

**Comment by neerajm20** [comment]
I am having the same issue and 'docker start reaction_mongo-init-replica_1' is not enough. When I log into mongo, there is no reaction database. What would be a workaround to get the system up and running?

**Comment by ticean** [comment]
Thanks @janus-reith. The problem is in the reaction project. The `mongo-init-replica` is launched in parallel with the `mongo` service by Docker Compose. Technically, there's a dependency defined in Compose,  but Compose makes no guarantee that `mongo` is ready at the time that `mongo-init-replica` is started. It results in a race condition.

The proper way to get around this is to get rid of the side-chain `mongo-init-replica`, and wrap `reaction` with a wait script. That could be done with an entrypoint, for example. That would force the replica set init script to run in series and thus reaction could be started once it's known that mongo is available and configured correctly.

So, in the `reaction` container:

1. Wait for `mongodb` to be ready
2. Setup the replica set. Gracefully handle when replicaset is already configured.
3. Start reaction.

**Comment by spencern** [comment]
This should be resolved by #4745
